examples/reading_events.rs

```diff
@@ -0,0 +1,188 @@
+
+    if let Some(mut events) = result.ok() {
+        while let Some(event) = events.try_next().await? {
+            let test_event = event.get_original_event().as_json::<TestEvent>()?;
+
+            println!("Event> {:?}", test_event);
+        }
+    }
+    // endregion reading-backwards
+
+    Ok(())
+}
+
+pub async fn read_from_all_stream(client: &Client) -> Result<()> {
+    // region read-from-all-stream
+    let mut events = client.read_all(&Default::default(), All).await?;
+    // endregion read-from-all-stream
+
+    // region read-from-all-stream-iterate
+    while let Some(event) = events.try_next().await? {
+        println!("Event> {:?}", event.get_original_event());
+    }
+    // endregion read-from-all-stream-iterate
+
+    Ok(())
+}
+
+pub async fn read_all_overriding_user_credentials(client: &Client) -> Result<()> {
+    // region read-from-all-stream
+    let options = ReadAllOptions::default().authenticated(Credentials::new("admin", "changeit"));
+    let events = client.read_all(&options, All).await?;
+    // endregion read-from-all-stream
+
+    Ok(())
+}
+
+pub async fn ignore_system_events(client: &Client) -> Result<()> {
+    // region ignore-system-events
+    let mut events = client.read_all(&Default::default(), All).await?;
+
+    while let Some(event) = events.try_next().await? {
+        if event.get_original_event().event_type.starts_with("$") {
+            continue;
+        }
+
+        println!("Event> {:?}", event.get_original_event());
+    }
+    // endregion ignore-system-events
+
+    Ok(())
+}
+
+pub async fn read_from_all_stream_backwards(client: &Client) -> Result<()> {
+    // region read-from-all-stream-backwards
+    let options = ReadAllOptions::default().origin(Origin::End);
+
+    let mut events = client.read_all(&options, All).await?;
+    // endregion read-from-all-stream-backwards
+
+    // region read-from-all-stream-iterate
+    while let Some(event) = events.try_next().await? {
+        println!("Event> {:?}", event.get_original_event());
+    }
+    // endregion read-from-all-stream-iterate
+
+    Ok(())
+}
+
+pub async fn filtering_out_system_events(client: &Client) -> Result<()> {
+    let mut events = client.read_all(&Default::default(), All).await?;
+
+    while let Some(event) = events.try_next().await? {
+        if !event.get_original_event().event_type.starts_with("$") {
+            continue;
+        }
+
+        println!("Event> {:?}", event.get_original_event());
+    }
+    Ok(())
+}
+
+pub async fn read_from_stream_resolving_link_tos(client: &Client) -> Result<()> {
+    // region read-from-all-stream-resolving-link-Tos
+    let options = ReadAllOptions::default().resolve_links();
+    let mut events = client.read_all(&options, All).await?;
+
+    while let Some(event) = events.try_next().await? {
+        println!("Event> {:?}", event.get_original_event());
+    }
+    // endregion read-from-all-stream-resolving-link-Tos
```

Comment:
**CONCERN:** Other samples are not showing iteration. Do you think that we should update other samples also to show getting the original event? Or we should not show it here?

Reply:
I found one client that does it. Maybe you get rid of it after your Java snippets PR. I'm fine not showing the iteration.

Reply:
As we're having now the changes, then I think we can hide iteration. I can handle the Java snippets update.

Reply:
done